src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateUtils.java

```diff
@@ -533,64 +464,44 @@ public static PodTemplate combine(PodTemplate parent, PodTemplate template) {
-                        ? template.getSlaveConnectTimeout()
-                        : parent.getSlaveConnectTimeout());
-
-        podTemplate.setIdleMinutes(
-                template.getIdleMinutes() != 0 ? template.getIdleMinutes() : parent.getIdleMinutes());
-
+        podTemplate.setInheritFrom(h.resolve(PodTemplate::getInheritFrom, PodTemplateUtils::isNullOrEmpty));
+        podTemplate.setInstanceCap(h.resolve(PodTemplate::getInstanceCap, i -> Objects.equals(i, Integer.MAX_VALUE)));
+        podTemplate.setSlaveConnectTimeout(h.resolve(
+                PodTemplate::getSlaveConnectTimeout,
+                i -> Objects.equals(i, PodTemplate.DEFAULT_SLAVE_JENKINS_CONNECTION_TIMEOUT)));
+        podTemplate.setIdleMinutes(h.resolve(PodTemplate::getIdleMinutes, i -> Objects.equals(i, 0)));
         podTemplate.setActiveDeadlineSeconds(
-                template.getActiveDeadlineSeconds() != 0
-                        ? template.getActiveDeadlineSeconds()
-                        : parent.getActiveDeadlineSeconds());
-
-        podTemplate.setServiceAccount(
-                !isNullOrEmpty(template.getServiceAccount())
-                        ? template.getServiceAccount()
-                        : parent.getServiceAccount());
-
-        podTemplate.setSchedulerName(
-                !isNullOrEmpty(template.getSchedulerName()) ? template.getSchedulerName() : parent.getSchedulerName());
-
+                h.resolve(PodTemplate::getActiveDeadlineSeconds, i -> Objects.equals(i, 0)));
+        podTemplate.setServiceAccount(h.resolve(PodTemplate::getServiceAccount, PodTemplateUtils::isNullOrEmpty));
+        podTemplate.setSchedulerName(h.resolve(PodTemplate::getSchedulerName, PodTemplateUtils::isNullOrEmpty));
         podTemplate.setPodRetention(template.getPodRetention());
-        podTemplate.setShowRawYaml(template.isShowRawYamlSet() ? template.isShowRawYaml() : parent.isShowRawYaml());
-
-        podTemplate.setRunAsUser(template.getRunAsUser() != null ? template.getRunAsUser() : parent.getRunAsUser());
-        podTemplate.setRunAsGroup(template.getRunAsGroup() != null ? template.getRunAsGroup() : parent.getRunAsGroup());
-
-        podTemplate.setSupplementalGroups(
-                template.getSupplementalGroups() != null
-                        ? template.getSupplementalGroups()
-                        : parent.getSupplementalGroups());
-
+        podTemplate.setShowRawYaml(h.resolve(PodTemplate::isShowRawYaml, v -> v));
+        podTemplate.setRunAsUser(h.resolve(PodTemplate::getRunAsUser, Objects::isNull));
+        podTemplate.setRunAsGroup(h.resolve(PodTemplate::getRunAsGroup, Objects::isNull));
+        podTemplate.setSupplementalGroups(h.resolve(PodTemplate::getSupplementalGroups, Objects::isNull));
+        podTemplate.setAgentContainer(h.resolve(PodTemplate::getAgentContainer, PodTemplateUtils::isNullOrEmpty));
+        podTemplate.setAgentInjection(h.resolve(PodTemplate::isAgentInjection, v -> template.isShowRawYamlSet()));
```

Comment:
```suggestion
```
?

Reply:
Ah, indeed, there is a copy/paste mistake here.